torch/_higher_order_ops/wrap.py

```diff
@@ -82,9 +82,16 @@ class TagActivationCheckpoint(HigherOrderOperator):
     in the forward and recomputed forward in backward.
     """
 
+    # Each utils.checkpoint call has an temporary entry in `context_fn_map` storing its context function.
+    # The entry is removed after the checkpointed function is traced through the first time
+    # (at which point the context function is moved to be attached to the graph module).
+    # Reason is that we want the lifetime of the context function to be tied to the graph module
+    # (because as long as the graph module is valid, the context function needs to stay valid),
+    # but there is no way to do this linking during Dynamo tracing, so we have to use this temp map as a bridge.
+    context_fn_tmp_map = {}
```

Comment:
@ezyang I updated it to be a temporary map (with rationale in comment) - wonder should I make this map TLS?

Reply:
Update: we don't have this global container anymore (and instead we directly attach the context_fn to the checkpointed graph module)

Reply:
outdated now right?